asyncer/_main.py

```diff
@@ -135,7 +135,7 @@ async def do_work(name: str) -> str:
         async with asyncer.create_task_group() as task_group:
             result1 = task_group.soonify(do_work)(name="task 1")
             result2 = task_group.soonify(do_work)(name="task 2")
-            await anyio.sleep(0)
+            await anyio.lowlevel.checkpoint()
```

Comment:
```suggestion
            await anyio.sleep(0)
```
I'm reverting this edit to focus on one topic per PR - that makes reviewing and discussion easier. If you'd like to add this change, please open a separate PR with it 🙏